src/components/Button.tsx

```diff
@@ -0,0 +1,29 @@
+import FancyRectangle from '@/components/FancyRectangle';
+import React from 'react';
+
+interface ButtonProps {
+    children: React.ReactNode;
+    colour: string;
+    href: string;
+}
+
+const Button = ({ children, colour, href }: ButtonProps) => {
+    const buttonColors: { [key: string]: string } = {
+        orange: 'bg-orange hover:bg-yellow',
+        purple: 'bg-purple hover:bg-yellow',
+    };
+
+    return (
+        <FancyRectangle colour="black" offset="4" filled={true}>
+            <a href={href}>
+                <button
+                    className={`whitespace-nowrap py-4 px-12 md:py-1 md:px-2 lg:py-2 lg:px-6 border-2 border-black font-bold transition-colors duration-300 ${buttonColors[colour]}`}
+                >
+                    {children}
+                </button>
+            </a>
+        </FancyRectangle>
```

Comment:
It's best not to nest a button inside a link tag, and vice versa. It's unclear on the intent (is it a button or a link?) and is [not valid HTML 5 according to the spec](https://stackoverflow.com/a/6393863).

[Here](https://www.buttoncheatsheet.com) is a handy cheat sheet of best practices when it comes to HTML buttons.